src/input/type_generic.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ * type_generic.cc
+ *
+ *  Created on: May 1, 2012
+ *      Author: jb
+ */
+
+
+#include <input/type_generic.hh>
+#include <input/type_repository.hh>
+
+#include <boost/functional/hash.hpp>
+
+
+
+namespace Input {
+
+namespace Type {
+
+
+/*******************************************************************
+ * implementation of Parameter
+ */
+
+Parameter::Parameter(const string & parameter_name)
+: name_(parameter_name) {}
+
+
+Parameter::Parameter(const Parameter & other)
+: name_(other.name_) {}
+
+
+string Parameter::type_name() const {
+    return name_;
+}
+
+
+TypeBase::TypeHash Parameter::content_hash() const {
+	TypeHash seed=0;
+	boost::hash_combine(seed, "Parameter");
+	boost::hash_combine(seed, type_name());
+
+	return seed;
+}
+
+
+bool Parameter::valid_default(const string &str) const {
```

Comment:
Where we need to call this method? This method should be used only in JSONToStorage which works only with IST where Parameter type should not appear. So, this method should throw always.